### PR TITLE
Update ServerTest.java, Fix Issue #2542

### LINKED ISSUE #2542: Bug in ServerTest.java, please fix
Hallo Uyuni-Team,
there is a Bug in ServerTest.java#L252:

[https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/domain/server/test/ServerTest.java#L252](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/domain/server/test/ServerTest.java#L252)
```
    /**
     * Test for {@link Server#doesOsSupportsOSImageBuilding()}.
     */
    public void testOsDoesNotSupportsOSImageBuilding() throws Exception {
       Server s = ServerFactoryTest.createTestServer(user, true,
                ServerConstants.getServerGroupTypeSaltEntitled(),
                ServerFactoryTest.TYPE_SERVER_MINION);
        s.setOs("SLES");
        s.setRelease("11.4");
        assertFalse(s.doesOsSupportsContainerization());
    }
```
The correct Lines 252 would be:
```
        assertFalse(s.doesOsSupportsOSImageBuilding());
```

Could you please correct the code.

Thanks in advance.

Best regards
Shirocco88

Hint: SLES11 does supports OSImageBuilding
https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/domain/server/MinionServer.java#L134

    /**
     * Return <code>true</code> if OS on this system supports OS Image building,
     * <code>false</code> otherwise.
     *
     * @return <code>true</code> if OS supports OS Image building
     */

    @Override
    public boolean doesOsSupportsOSImageBuilding() {
        return isSLES11() || isSLES12() || isSLES15() || isLeap15();
    }

### PR DESCRIPTION
**Pull request by Shirocco88**
## What does this PR change?
Fix Issue #2542
**add description**

## Documentation
- No documentation needed

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes #2542 

- [ ] **DONE**

## Changelogs

- [X] No changelog needed

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/domain/server/test/ServerTest.java [hunk 1]
```diff
@@ -248,8 +248,8 @@ public void testOsDoesNotSupportsOSImageBuilding() throws Exception {
                 ServerConstants.getServerGroupTypeSaltEntitled(),
                 ServerFactoryTest.TYPE_SERVER_MINION);
         s.setOs("SLES");
-        s.setRelease("11.4");
-        assertFalse(s.doesOsSupportsContainerization());
+        s.setRelease("10.4");
```
Comment (juliogonzalez) [inline java/code/src/com/redhat/rhn/domain/server/test/ServerTest.java:251]: I don't understand, on your issue you said:

> Hint: SLES11 does supports OSImageBuilding

But here you are changing to SLE10.

Can you explain?

**Comment by Shirocco88** [comment]
@juliogonzalez: do you need further inputs to fix the issue ?

**Comment by juliogonzalez** [comment]
@Shirocco88 I left a question. I am not a Java guy, but I am pretty sure the future reviewer will have the same one.

**Comment by Shirocco88** [comment]
SLES11-SP4 as well as SLES12-SP1 are supported/enabled for OSImageBuilding;
my intension was, use existing function name testOsDoesNotSupportsOSImageBuilding()

Starting with reference:
https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/domain/server/test/ServerTest.java#L162
```
    public void testOsSupportsOSImageBuilding() throws Exception {
SNIP
        s.setOs("SLES");
        s.setRelease("12.1");
        assertTrue(s.doesOsSupportsOSImageBuilding());
```

https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/domain/server/test/ServerTest.java#L246
```
    public void testOsDoesNotSupportsOSImageBuilding() throws Exception {
SNIP
    s.setRelease("11.4");
    assertFalse(s.doesOsSupportsOSImageBuilding());
```

@juliogonzalez: you are right; changing L252 is sufficient;
sorry for mismatch
I'll delete my wrong comments

**Comment by Shirocco88** [comment]
@juliogonzalez: Check "java_pgsql_tests" is Pending
what can I do to run the checks completely/successfully ?

**Comment by juliogonzalez** [comment]
Don't worry, it's still running, and should get the status updated soon.

We still need to make the CI public, so public contributors can see the result. I will paste the results if there are failures.

**Comment by juliogonzalez** [comment]
I am removing myself as reviewer, as in the end I am not part of the java development teams. Already asked a review from them.

**Comment by juliogonzalez** [comment]
The test failed:
```
junit.framework.AssertionFailedError
	at com.redhat.rhn.domain.server.test.ServerTest.testOsDoesNotSupportsOSImageBuilding(ServerTest.java:252)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```
To be honest, I fail to understand why this PR is needed, but I think it's better if one of our Java developers have a look :-)

**Comment by aaannz** [comment]
> @juliogonzalez: you are right; changing L252 is sufficient;
sorry for mismatch;
I'll delete my wrong comments.

On the contrary, your initial change was correct. You are testing that os image building is NOT supported. So you need to pass os version where it is not supported -> 10.x.
Nice catch about the wrong test ( `doesOsSupportsContainerization` vs `doesOsSupportsOSImageBuilding` ) I was wondering how that test could pass before.

**Comment by aaannz** [comment]
Can you also squash the commits to one, please?

**Review by aaannz — APPROVED** [review]
Thanks for the contribution.

**Comment by Shirocco88** [comment]
@aaannz: 
I pushed the commit 16 hours ago, but there is still one check "java-pgsql-test" in pending state in pipeline; 
what can I do to run the checks completely/successfully ?

**Comment by aaannz** [comment]
Details of the job are internal, however seems to me now that it is running and started around 3 minutes ago.

**Comment by aaannz** [comment]
At the end the issue was that nothing (I suppose it was a gittaro job) reset checkboxes in the PR. The test was running again and again and again, always ended successfully and immediately restarted. Once I unchecked those checkboxes test finished and correct result was set.

**Comment by juliogonzalez** [comment]
@aaannz, please consider if this needs to be backported.

**Pull request merged**